### PR TITLE
mgr/DaemonServer: [Cleanup] Remove redundant code

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
test this please

File: src/mgr/DaemonServer.cc [hunk 1]
```diff
@@ -665,29 +665,27 @@ bool DaemonServer::handle_command(MCommand *m)
...
-      dout(1) << " access denied" << dendl;
-      ss << "access denied; does your client key have mgr caps?"
-	" See http://docs.ceph.com/docs/master/mgr/administrator/#client-authentication";
-      cmdctx->reply(-EACCES, ss);
-      return true;
-    }
-  } else {
+    is_allowed = _allowed_command(session.get(), py_command.module,
+      prefix, cmdctx->cmdmap, param_str_map, &py_command);
+  } else
```
Comment (liewegas) [inline src/mgr/DaemonServer.cc]: braces around else segment, please

**Comment by shinobu-x** [comment]
@liewegas Please check when you get a chance.

File: src/mgr/DaemonServer.cc [hunk 2]
```diff
@@ -665,29 +665,27 @@ bool DaemonServer::handle_command(MCommand *m)
...
+      prefix, cmdctx->cmdmap,  param_str_map, mgr_cmd);
+  }
+  if (!is_allowed) {
       dout(1) << " access denied" << dendl;
       audit_clog->info() << "from='" << session->inst << "' "
                          << "entity='" << session->entity_name << "' "
                          << "cmd=" << m->cmd << ":  access denied";
-      ss << "access denied' does your client key have mgr caps?"
-	" See http://docs.ceph.com/docs/master/mgr/administrator/#client-authentication";
+      ss << "access denied' does your client key have mgr caps? "
```
Comment (tchaikov) [inline src/mgr/DaemonServer.cc]: i don't think the single quote is expected here. might want to use period or a semicolon. it's a user facing message. that's why i am picky here.

@jcsp what do you think?

**Comment by tchaikov** [comment]
@shinobu-x please remove the infix of "[cleanup]" from the title of the commit message.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-12-06_15:52:24-rados-wip-kefu-testing-2017-12-06-1717-distro-basic-mira/

failures tracked by: 
- http://tracker.ceph.com/issues/22349
- http://tracker.ceph.com/issues/22093

File: src/mgr/DaemonServer.cc (hunk 2 above)
Comment (jcsp) [reply src/mgr/DaemonServer.cc]: yes, the string is wrong but it's not being changed by this PR so I'll just clean it up separately

**Pull request merged**